### PR TITLE
Feat(web-react): UNSTABLE_Slider #DS-1322

### PR DESCRIPTION
**Pull request by pavelklibani**
### Issue reference

[Slider | web-react (UNSTABLE)](https://jira.almacareer.tech/browse/DS-1322)

### CODE REVIEW

File: packages/web-react/src/components/UNSTABLE_Slider/demo/SliderDefault.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React, { useState } from 'react';
+import { DEMO_SLIDER_DEFAULT_VALUE } from '../constants';
+import UNSTABLE_Slider from '../UNSTABLE_Slider';
+
+const SliderDefault = () => {
+  const [value, setValue] = useState(DEMO_SLIDER_DEFAULT_VALUE);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (literat) [inline packages/web-react/src/components/UNSTABLE_Slider/demo/SliderDefault.tsx:8]:
```suggestion
  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
```

File: packages/web-react/src/components/UNSTABLE_Slider/demo/SliderFluid.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React, { useState } from 'react';
+import { DEMO_SLIDER_DEFAULT_VALUE } from '../constants';
+import UNSTABLE_Slider from '../UNSTABLE_Slider';
+
+const SliderFluid = () => {
+  const [value, setValue] = useState(DEMO_SLIDER_DEFAULT_VALUE);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (literat) [inline packages/web-react/src/components/UNSTABLE_Slider/demo/SliderFluid.tsx:8]:
```suggestion
  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
```

File: packages/web-react/src/components/UNSTABLE_Slider/demo/SliderHelperText.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React, { useState } from 'react';
+import { DEMO_SLIDER_DEFAULT_VALUE } from '../constants';
+import UNSTABLE_Slider from '../UNSTABLE_Slider';
+
+const SliderHelperText = () => {
+  const [value, setValue] = useState(DEMO_SLIDER_DEFAULT_VALUE);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (literat) [inline packages/web-react/src/components/UNSTABLE_Slider/demo/SliderHelperText.tsx:8]:
```suggestion
  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
```

File: packages/web-react/src/components/UNSTABLE_Slider/demo/SliderHiddenLabel.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React, { useState } from 'react';
+import { DEMO_SLIDER_DEFAULT_VALUE } from '../constants';
+import UNSTABLE_Slider from '../UNSTABLE_Slider';
+
+const SliderHiddenLabel = () => {
+  const [value, setValue] = useState(DEMO_SLIDER_DEFAULT_VALUE);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (literat) [inline packages/web-react/src/components/UNSTABLE_Slider/demo/SliderHiddenLabel.tsx:8]:
```suggestion
  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
```

File: packages/web-react/src/components/UNSTABLE_Slider/demo/SliderRequired.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React, { useState } from 'react';
+import { DEMO_SLIDER_DEFAULT_VALUE } from '../constants';
+import UNSTABLE_Slider from '../UNSTABLE_Slider';
+
+const SliderRequired = () => {
+  const [value, setValue] = useState(DEMO_SLIDER_DEFAULT_VALUE);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (literat) [inline packages/web-react/src/components/UNSTABLE_Slider/demo/SliderRequired.tsx:8]:
```suggestion
  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
```

File: packages/web-react/src/components/UNSTABLE_Slider/demo/SliderSelected.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import React, { useState } from 'react';
+import { DEMO_SLIDER_DEFAULT_VALUE } from '../constants';
+import UNSTABLE_Slider from '../UNSTABLE_Slider';
+
+const SliderSelected = () => {
+  const [value, setValue] = useState(DEMO_SLIDER_DEFAULT_VALUE);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (literat) [inline packages/web-react/src/components/UNSTABLE_Slider/demo/SliderSelected.tsx:8]:
```suggestion
  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
```

File: packages/web-react/src/components/UNSTABLE_Slider/demo/SliderValidation.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react';
+import { DEMO_SLIDER_DEFAULT_VALUE } from '../constants';
+import UNSTABLE_Slider from '../UNSTABLE_Slider';
+
+const SliderValidation = () => {
+  const [valueSuccess, setValueSuccess] = useState(DEMO_SLIDER_DEFAULT_VALUE);
+  const [valueWarning, setValueWarning] = useState(DEMO_SLIDER_DEFAULT_VALUE);
+  const [valueDanger, setValueDanger] = useState(DEMO_SLIDER_DEFAULT_VALUE);
+
+  const handleChangeSuccess = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (literat) [inline packages/web-react/src/components/UNSTABLE_Slider/demo/SliderValidation.tsx:10]:
```suggestion
  const handleChangeSuccess = (event: ChangeEvent<HTMLInputElement>) => {
```

Comment (literat) [reply packages/web-react/src/components/UNSTABLE_Slider/demo/SliderValidation.tsx:10]: And everywhere else, please.

File: packages/web-react/src/components/UNSTABLE_Slider/useSlider.ts
Comment (literat) [inline packages/web-react/src/components/UNSTABLE_Slider/useSlider.ts:1]: Are you sure about the naming? This hook looks too general to be just named `useSlider`. From my point of view, it can be used for any input to set the value when changed.

File: packages/web-react/src/components/UNSTABLE_Slider/demo/SliderValidation.tsx (hunk 7 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/demo/SliderValidation.tsx]: fixed

File: packages/web-react/src/components/UNSTABLE_Slider/README.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+| `isFluid`        | `bool`                   | `false` | ✕        | Whether the element spans to the full width of its parent |
+| `isSelected`     | `bool`                   | `false` | ✕        | Whether is the item selected                              |
+| `label`          | `string`                 | -       | ✓        | Label text                                                |
+| `max`            | `number`                 | 100     | ✕        | Max value of slider input                                 |
+| `min`            | `number`                 | 0       | ✕        | Min value of slider input                                 |
+| `onChange`       | `() => void`             | -       | ✓        | On input change callback                                  |
+| `step`           | `number`                 | 1       | ✕        | Sets the stepping interval                                |
+| `validationText` | [`string` \| `string[]`] | -       | ✕        | Validation text                                           |
+| `value`          | `number`                 | 0       | ✓        | Input value                                               |
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Slider/README.md:35]: required and default? is it correct?

File: packages/web-react/src/components/UNSTABLE_Slider/README.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+<UNSTABLE_Slider id="slider" label="Slider" value={value} onChange={handleChange} />;
+```
+
+### API
+
+| Attribute        | Type                     | Default | Required | Description                                               |
+| ---------------- | ------------------------ | ------- | -------- | --------------------------------------------------------- |
+| `helperText`     | `string`                 | -       | ✕        | Custom helper text                                        |
+| `id`             | `string`                 | -       | ✓        | Input and label identification                            |
+| `isFluid`        | `bool`                   | `false` | ✕        | Whether the element spans to the full width of its parent |
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Slider/README.md:27]: I am missing isLabelHidden (or how do we call it in other components)

isRequired, isDisabled

File: packages/web-react/src/components/UNSTABLE_Slider/README.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+| `isSelected`     | `bool`                   | `false` | ✕        | Whether is the item selected                              |
+| `label`          | `string`                 | -       | ✓        | Label text                                                |
+| `max`            | `number`                 | 100     | ✕        | Max value of slider input                                 |
+| `min`            | `number`                 | 0       | ✕        | Min value of slider input                                 |
+| `onChange`       | `() => void`             | -       | ✓        | On input change callback                                  |
+| `step`           | `number`                 | 1       | ✕        | Sets the stepping interval                                |
+| `validationText` | [`string` \| `string[]`] | -       | ✕        | Validation text                                           |
+| `value`          | `number`                 | 0       | ✓        | Input value                                               |
+
+The components accept [additional attributes][readme-additional-attributes].
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Slider/README.md:37]:
```suggestion
On top of the API options, the components accept [additional attributes][readme-additional-attributes].
If you need more control over the styling of a component, you can use [style props][readme-style-props]
and [escape hatches][readme-escape-hatches].
```

File: packages/web-react/src/components/UNSTABLE_Slider/README.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+| `isFluid`        | `bool`       | `false` | ✕        | Whether the element spans to the full width of its parent |
+| `isSelected`     | `bool`       | `false` | ✕        | Whether is the item selected                              |
+| `label`          | `string`     | -       | ✓        | Label text                                                |
+| `max`            | `number`     | 100     | ✕        | Max value of slider input                                 |
+| `min`            | `number`     | 0       | ✕        | Min value of slider input                                 |
+| `onChange`       | `() => void` | -       | ✓        | On input change callback                                  |
+| `step`           | `number`     | 1       | ✕        | Sets the stepping interval                                |
+| `validationText` | `string`     | -       | ✕        | Validation text                                           |
+| `value`          | `number`     | 0       | ✕        | Input value                                               |
+
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Slider/README.md:66]:
```suggestion
On top of the API options, the components accept [additional attributes][readme-additional-attributes].
If you need more control over the styling of a component, you can use [style props][readme-style-props]
and [escape hatches][readme-escape-hatches].

```

File: packages/web-react/src/components/UNSTABLE_Slider/README.md [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+| `id`             | `string`     | -       | ✓        | Input and label identification                            |
+| `isFluid`        | `bool`       | `false` | ✕        | Whether the element spans to the full width of its parent |
+| `isSelected`     | `bool`       | `false` | ✕        | Whether is the item selected                              |
+| `label`          | `string`     | -       | ✓        | Label text                                                |
+| `max`            | `number`     | 100     | ✕        | Max value of slider input                                 |
+| `min`            | `number`     | 0       | ✕        | Min value of slider input                                 |
+| `onChange`       | `() => void` | -       | ✓        | On input change callback                                  |
+| `step`           | `number`     | 1       | ✕        | Sets the stepping interval                                |
+| `validationText` | `string`     | -       | ✕        | Validation text                                           |
+| `value`          | `number`     | 0       | ✕        | Input value                                               |
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Slider/README.md:65]: here the value is not required?

File: packages/web-react/src/components/UNSTABLE_Slider/README.md [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+
+### API
+
+| Attribute        | Type         | Default | Required | Description                                               |
+| ---------------- | ------------ | ------- | -------- | --------------------------------------------------------- |
+| `helperText`     | `string`     | -       | ✕        | Custom helper text                                        |
+| `id`             | `string`     | -       | ✓        | Input and label identification                            |
+| `isFluid`        | `bool`       | `false` | ✕        | Whether the element spans to the full width of its parent |
+| `isSelected`     | `bool`       | `false` | ✕        | Whether is the item selected                              |
+| `label`          | `string`     | -       | ✓        | Label text                                                |
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Slider/README.md:59]: I am missing the same props here

File: packages/web-react/src/components/UNSTABLE_Slider/UNSTABLE_Slider.tsx [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+    'aria-describedby': ariaDescribedBy,
+    helperText,
+    id,
+    label,
+    validationState,
+    validationText,
+    value,
+    min,
+    max,
+    step,
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Slider/UNSTABLE_Slider.tsx:27]: I don't know if we sort these alphabetically.

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+import { stylePropsTest } from '../../../../tests/providerTests/stylePropsTest';
+import UNSTABLE_Slider from '../UNSTABLE_Slider';
+
+describe('UNSTABLE_Slider', () => {
+  const id = 'slider-test';
+  const label = 'Slider';
+  const onChange = () => {};
+  const defaultValue = 10;
+
+  stylePropsTest((props) => <UNSTABLE_Slider id={id} {...props} data-testid="tooltip-test" />, 'tooltip-test');
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx:14]: tooltip :)

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+
+  restPropsTest((props) => <UNSTABLE_Slider id={id} {...props} />, 'div');
+
+  it('should render tooltip', () => {
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx:18]: tooltip :)

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_UncontrolledSlider.tsx [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+import { restPropsTest } from '../../../../tests/providerTests/restPropsTest';
+import { stylePropsTest } from '../../../../tests/providerTests/stylePropsTest';
+import UNSTABLE_UncontrolledSlider from '../UNSTABLE_UncontrolledSlider';
+
+describe('UNSTABLE_Slider', () => {
+  const id = 'slider-test';
+  const label = 'Slider';
+
+  stylePropsTest(
+    (props) => <UNSTABLE_UncontrolledSlider id={id} {...props} data-testid="tooltip-test" />,
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_UncontrolledSlider.tsx:13]: tooltip :)

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_UncontrolledSlider.tsx [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+    'tooltip-test',
+  );
+
+  restPropsTest((props) => <UNSTABLE_UncontrolledSlider id={id} {...props} />, 'div');
+
+  it('should render tooltip', () => {
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_UncontrolledSlider.tsx:19]: tooltip :)

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+        id={id}
+        value={defaultValue}
+        onChange={onChange}
+        label={label}
+        validationText="Validation text"
+        validationState="danger"
+      />,
+    );
+
+    expect(screen.getByText('Validation text')).toBeInTheDocument();
```
Comment (crishpeen) [inline packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx:46]: isn't there a role?

File: packages/web-react/src/components/UNSTABLE_Slider/README.md (hunk 8 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/README.md:35]: fixed

File: packages/web-react/src/components/UNSTABLE_Slider/README.md (hunk 9 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/README.md:27]: `isLabelHidden` and `isRequired` have been added to the table. We don't have `isDisabled` yet, but it will be added in the future.

File: packages/web-react/src/components/UNSTABLE_Slider/README.md (hunk 10 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/README.md:37]: changed

File: packages/web-react/src/components/UNSTABLE_Slider/README.md (hunk 11 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/README.md:66]: added

File: packages/web-react/src/components/UNSTABLE_Slider/README.md (hunk 12 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/README.md:65]: Here it is not necessary, because the state of the uncontrolled value is managed by our hook. In a controlled component, the value must be provided by the developer.

File: packages/web-react/src/components/UNSTABLE_Slider/README.md (hunk 13 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/README.md:59]: added

File: packages/web-react/src/components/UNSTABLE_Slider/UNSTABLE_Slider.tsx (hunk 14 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/UNSTABLE_Slider.tsx:27]: why not, sorted

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx (hunk 15 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx:14]: 🙈

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx (hunk 19 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx:46]: A can't find any role for this element

Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx:46]: There is no role

File: packages/web-react/src/components/UNSTABLE_Slider/useSlider.ts
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/useSlider.ts:1]: we agreed to add a comment about the need to refactor the hook if no changes are made in the future

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+import React from 'react';
+import { restPropsTest } from '../../../../tests/providerTests/restPropsTest';
+import { stylePropsTest } from '../../../../tests/providerTests/stylePropsTest';
+import UNSTABLE_Slider from '../UNSTABLE_Slider';
+
+describe('UNSTABLE_Slider', () => {
+  const id = 'slider-test';
+  const label = 'Slider';
+  const onChange = () => {};
+  const defaultValue = 10;
```
Comment (curdaj) [inline packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx:12]:
```suggestion
  const DEFAULT_VALUE = 10;
```
?

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_UncontrolledSlider.tsx [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+  );
+
+  restPropsTest((props) => <UNSTABLE_UncontrolledSlider id={id} {...props} />, 'div');
+
+  it('should render slider', () => {
+    render(<UNSTABLE_UncontrolledSlider id={id} label={label} />);
+
+    const sliderElement = screen.getByRole('slider');
+
+    expect(screen.getByText(label)).toBeInTheDocument();
```
Comment (curdaj) [inline packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_UncontrolledSlider.tsx:24]:
```suggestion
    expect(screen.getByLabelText(label)).toBeInTheDocument();
```

**Review by curdaj — APPROVED** [review]

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_UncontrolledSlider.tsx (hunk 21 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_UncontrolledSlider.tsx]: changed, thank you

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx (hunk 20 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx:12]: I looked in other test files and we have constants in camelCase, so I'll leave it like this.

File: exporters/js/generated/functions.js
Comment (adamkudrna) [inline exporters/js/generated/functions.js:1]: Why the changes in the exporters…?

File: packages/web-react/src/components/UNSTABLE_Slider/README.md [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+| `helperText`     | `string`     | -       | ✕        | Custom helper text                                        |
+| `id`             | `string`     | -       | ✓        | Input and label identification                            |
+| `isFluid`        | `bool`       | `false` | ✕        | Whether the element spans to the full width of its parent |
+| `isLabelHidden`  | `bool`       | `false` | ✕        | Whether the label is hidden                               |
+| `isRequired`     | `bool`       | `false` | ✕        | Whether is the input required                             |
+| `isSelected`     | `bool`       | `false` | ✕        | Whether is the input selected                             |
+| `label`          | `string`     | -       | ✓        | Label text                                                |
+| `max`            | `number`     | 100     | ✕        | Max value of slider input                                 |
+| `min`            | `number`     | 0       | ✕        | Min value of slider input                                 |
+| `onChange`       | `() => void` | -       | ✓        | On input change callback                                  |
```
Comment (adamkudrna) [inline packages/web-react/src/components/UNSTABLE_Slider/README.md:66]: If I get it right, `onChange` is not required for the uncontrolled `Slider`. At least it's not in your example:

```jsx
<UNSTABLE_UncontrolledSlider id="slider-uncontrolled" label="UncontrolledSlider" />
```

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+
+  stylePropsTest((props) => <UNSTABLE_Slider id={id} {...props} data-testid="slider-test" />, 'slider-test');
+
+  restPropsTest((props) => <UNSTABLE_Slider id={id} {...props} />, 'div');
+
+  it('should render slider', () => {
+    render(<UNSTABLE_Slider id={id} value={defaultValue} onChange={onChange} label={label} />);
```
Comment (adamkudrna) [inline packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx:19]: You could have it like this:

```jsx
const defaultProps = {
  id: 'slider-test',
  label: 'Slider',
  onChange: () => {},
  value: defaultValue,
}

<UNSTABLE_Slider {...defaultProps} />
```

And later on in other tests:

```jsx
<UNSTABLE_Slider {...defaultProps} helperText='Helper text' />
```

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+
+    const sliderElement = screen.getByRole('slider');
+
+    expect(screen.getByText(label)).toBeInTheDocument();
+    expect(sliderElement).toBeInTheDocument();
```
Comment (adamkudrna) [inline packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx:24]: Isn't this test implicitly true when `screen.getByRole('slider')` does not fail?

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+        id={id}
+        value={defaultValue}
+        onChange={onChange}
+        label={label}
+        validationText={validationText}
+        validationState="danger"
+      />,
+    );
+    screen.debug();
+    expect(screen.getByText(validationText)).toBeInTheDocument();
```
Comment (adamkudrna) [inline packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx:50]: No test of the CSS class as in the uncontrolled variant?

File: packages/web-react/src/components/UNSTABLE_Slider/constants.ts [hunk 26]
```diff
@@ -0,0 +1,7 @@
+export const DEMO_SLIDER_DEFAULT_VALUE = 30;
+
+export const SLIDER_DEFAULT_PROPS = {
+  max: 100, // @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range#specifying_the_minimum_and_maximum
+  min: 0, // @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range#specifying_the_minimum_and_maximum
+  step: 1, // @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range#setting_the_values_granularity
```
Comment (adamkudrna) [inline packages/web-react/src/components/UNSTABLE_Slider/constants.ts:6]: 👍🏻

File: packages/web-react/src/components/UNSTABLE_Slider/demo/UncontrolledSlider.tsx
Comment (adamkudrna) [inline packages/web-react/src/components/UNSTABLE_Slider/demo/UncontrolledSlider.tsx:1]: This demo isn't used anywhere, is it right?

File: packages/web-react/src/components/UNSTABLE_Slider/stories/UNSTABLE_Slider.stories.tsx [hunk 27]
```diff
@@ -0,0 +1,149 @@
...
+    min: {
+      control: 'number',
+      table: {
+        defaultValue: { summary: 0 },
+      },
+    },
+    step: {
+      control: 'number',
+      table: {
+        defaultValue: { summary: 1 },
```
Comment (adamkudrna) [inline packages/web-react/src/components/UNSTABLE_Slider/stories/UNSTABLE_Slider.stories.tsx:76]: Shouldn't we reuse the default values defined in types?

File: packages/web-react/src/components/UNSTABLE_Slider/stories/UNSTABLE_Slider.stories.tsx [hunk 28]
```diff
@@ -0,0 +1,149 @@
...
+    helperText: '',
+    id: 'slider',
+    isFluid: false,
+    isLabelHidden: false,
+    isRequired: false,
+    isSelected: false,
+    label: 'Label',
+    max: 100,
+    min: 0,
+    step: 1,
```
Comment (adamkudrna) [inline packages/web-react/src/components/UNSTABLE_Slider/stories/UNSTABLE_Slider.stories.tsx:109]: Shouldn't we reuse the default values defined in types?

**Review by adamkudrna — APPROVED** [review]
Very nice beautiful! 👏🏻

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx (hunk 23 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx:19]: good idea!

File: packages/web-react/src/components/UNSTABLE_Slider/demo/UncontrolledSlider.tsx
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/demo/UncontrolledSlider.tsx:1]: Yes, we are not displaying uncontrolled demos because they are only in React and would break visual tests. However, I leave them there for when we need to test or debug something, so they are ready to add and test.

File: packages/web-react/src/components/UNSTABLE_Slider/stories/UNSTABLE_Slider.stories.tsx (hunk 27 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/stories/UNSTABLE_Slider.stories.tsx:76]: imported from constants 👍

File: packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx (hunk 24 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx:30]: i think i don't understand the question

Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/__tests__/UNSTABLE_Slider.test.tsx:30]: we discussed it and we leave it like this

File: packages/web-react/src/components/UNSTABLE_Slider/UNSTABLE_Slider.tsx [hunk 29]
```diff
@@ -0,0 +1,84 @@
...
+
+  const { classProps, props: modifiedProps } = useSliderStyleProps({ ...restProps, validationState });
+  const { styleProps, props: otherProps } = useStyleProps(modifiedProps);
+  const [ids, register] = useAriaIds(ariaDescribedBy);
+
+  const CSSVariable = '--slider-position';
+
+  const getSliderPosition = (num: number) => `${Math.round((100 * num) / 100)}%`;
+
+  const handleInput = (event: React.FormEvent<HTMLInputElement>) => {
```
Comment (literat) [inline packages/web-react/src/components/UNSTABLE_Slider/UNSTABLE_Slider.tsx:39]:
```suggestion
  const handleInput = (event: FormEvent<HTMLInputElement>) => {
```

File: packages/web-react/src/components/UNSTABLE_Slider/UNSTABLE_Slider.tsx [hunk 30]
```diff
@@ -0,0 +1,84 @@
...
+    const { target } = event as React.ChangeEvent<HTMLInputElement>;
```
Comment (literat) [inline packages/web-react/src/components/UNSTABLE_Slider/UNSTABLE_Slider.tsx:40]:
```suggestion
    const { target } = event as ChangeEvent<HTMLInputElement>;
```

File: packages/web-react/src/components/UNSTABLE_Slider/UNSTABLE_Slider.tsx [hunk 31]
```diff
@@ -0,0 +1,84 @@
...
+      <label htmlFor={id} className={classProps.label}>
+        {label}
+      </label>
+      <input
+        aria-describedby={ids.join(' ')}
+        className={classProps.input}
+        id={id}
+        onInput={handleInput}
+        ref={ref}
+        style={{ [CSSVariable]: `${getSliderPosition(value)}` } as React.CSSProperties}
```
Comment (literat) [inline packages/web-react/src/components/UNSTABLE_Slider/UNSTABLE_Slider.tsx:57]:
```suggestion
        style={{ [CSSVariable]: `${getSliderPosition(value)}` } as CSSProperties}
```

File: packages/web-react/src/components/UNSTABLE_Slider/UNSTABLE_Slider.tsx [hunk 32]
```diff
@@ -0,0 +1,84 @@
...
+        type="range"
+        min={min}
+        max={max}
+        step={step}
+        value={value}
+      />
+      <HelperText
```
Comment (literat) [inline packages/web-react/src/components/UNSTABLE_Slider/UNSTABLE_Slider.tsx:64]: Is the HelperText mandatory? The docs say the opposite.

File: packages/web-react/src/components/UNSTABLE_Slider/useSlider.ts [hunk 33]
```diff
@@ -0,0 +1,24 @@
+import { useState } from 'react';
+
+export interface UseSliderProps {}
+
+export interface UseSliderReturn {
+  value: number;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
```
Comment (literat) [inline packages/web-react/src/components/UNSTABLE_Slider/useSlider.ts:7]:
```suggestion
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
```

File: packages/web-react/src/components/UNSTABLE_Slider/useSlider.ts [hunk 34]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+// TODO: If in the future there is no need to expand this hook, we can refactor it and make it a general hook.
+// @see: https://designcode.io/react-hooks-handbook-useinput-hook
+
+export const useSlider = (propsValue: number): UseSliderReturn => {
+  const [value, setValue] = useState(propsValue);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (literat) [inline packages/web-react/src/components/UNSTABLE_Slider/useSlider.ts:16]:
```suggestion
  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
```

File: packages/web-react/src/components/UNSTABLE_Slider/UNSTABLE_Slider.tsx (hunk 32 above)
Comment (pavelklibani) [reply packages/web-react/src/components/UNSTABLE_Slider/UNSTABLE_Slider.tsx:64]: Helper text is not mandatory, the check for the prop is inside this component.
